great_expectations/rule_based_profiler/helpers/util.py

```diff
@@ -1080,3 +1091,17 @@ def sanitize_parameter_name(
         name = f"{name}{FULLY_QUALIFIED_PARAMETER_NAME_SEPARATOR_CHARACTER}{suffix}"
 
     return name.replace(FULLY_QUALIFIED_PARAMETER_NAME_SEPARATOR_CHARACTER, "_")
+
+
+class _NumericIterableWithDtype(Iterable, Protocol):
+    @property
+    def dtype(self) -> Any:
+        ...
+
+
+def _is_iterable_numeric_and_has_dtype_attr(
+    obj: Any,
+) -> TypeGuard[_NumericIterableWithDtype]:
+    if hasattr(obj, "dtype") and not np.issubdtype(obj.dtype, np.number):
```

Comment:
```suggestion
    if hasattr(obj, "dtype") and np.issubdtype(obj.dtype, np.number):
```
Shouldn't both these be true?